Optimisation: reduce code size in showing file name being printed

gudnimg opened the pull request:
The PR optimizes how the SD card file name is printed on the screen.

`lcd_print_pad()` now returns the last character it's pointer pointed to which is `'\0'` (NULL) when the end of the string was reached, or a non-zero value which indicates the opposite. We can use this directly to make the file name scroll to the left on the LCD by shifting the starting pointer by one index each time `lcd_print_pad()` didn't reach the end of the string.

Affected UI:
* SD card filename on the status screen
* SD card filenames in the "Print from SD" menu.

Change in memory:
Flash: -140 bytes
SRAM: 0 bytes

gudnimg committed:
optimisation: reduce code size in showing file name being printed

Change in memory:
Flash: -52 bytes
SRAM: 0 bytes

gudnimg commented:
Note: I have not tested this with a SD file which has a name length less than `LCD_WIDTH`. Will hopefully test that later today.

leptun reviewed (commented):
I feel like we can optimize this much further. We just need a way of knowing if lcd_print_pad got to the end of the string or not. I propose that lcd_print_pad `return *s;` so that we can check if we reached the end of the string or not. That would simplify this whole filename printing on the status screen. The same optimization could be applied also to the SD menu (maybe)

gudnimg committed:
further optimise lcdui_print_status_line

We can use lcd_print_pad to determine
whether the end of the file name was reached

lcd_print_pad now returns the last character pointed to.
If the end of the string was reached, it will be the null delimeter which
evaluates to "false" in an if statment (same as NULL).
Else "true" means the end of the string was not reached.

Change in memory:
Flash: -42 bytes
SRAM: 0 bytes

gudnimg commented:
Marking as ready for review since I tested both short filenames and long filenames. Both work without issues.

leptun reviewed (approved):
This looks perfect. Can't test it though, so I take your word that it works.
The next step might be to make the SD menu printing similar to this. It should be quite similar to this in the way it's rendered (- the folder icon and the cursor).

gudnimg committed:
optimise lcd_sdcard_menu with lcd_print_pad

Change in memory:
Flash: -46 bytes
SRAM: 0 bytes

leptun reviewed (approved):
Tested on MK404. Reapproving
